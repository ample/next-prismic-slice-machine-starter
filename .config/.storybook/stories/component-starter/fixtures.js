export default {
  componentProps: {
    nameOfProp: 'name of prop content goes here.',
  },
  themeOption: {
    nameOfProp: 'name of prop content goes here.',
    theme: 'theme-option',
  },
};
