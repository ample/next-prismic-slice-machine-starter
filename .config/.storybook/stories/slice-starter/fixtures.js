// ---------------------------------------------------------

export default {
  _props: {
    nameOfProp: [
      {
        spans: [],
        text: 'name of prop content goes here.',
        type: 'heading2',
      },
    ],
  },
  variationOption: {
    nameOfProp: [
      {
        spans: [],
        text: 'name of prop content goes here.',
        type: 'heading2',
      },
    ],
    variation: 'variation-option',
  },
}
