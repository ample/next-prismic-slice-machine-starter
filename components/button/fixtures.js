const buttonProps = {
  children: 'Call to action',
  theme: 'arrow',
  url: '#',
};

// ---------------------------------------------------------

export default {
  componentProps: {
    ...buttonProps,
  },
  themeArrow: {
    ...buttonProps,
  },
  themeDefault: {
    ...buttonProps,
    theme: 'default',
  },
  themeOutline: {
    ...buttonProps,
    theme: 'outline',
  },
  withButtonElement: {
    ...buttonProps,
    theme: 'default',
    type: 'button',
    url: null,
  },
  withButtonElementDisabled: {
    ...buttonProps,
    disabled: true,
    theme: 'default',
    type: 'button',
    url: null,
  },
};
