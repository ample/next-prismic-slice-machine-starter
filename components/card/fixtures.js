// ---------------------------------------------------------

const card = {
  body: [
    {
      spans: [],
      text: 'Est proident adipisicing reprehenderit in occaecat non. Aute enim minim labore nisi ullamco enim sint Lorem excepteur exercitation anim tempor ad minim ut. Sunt enim aliqua elit occaecat.',
      type: 'paragraph',
    },
  ],
  heading: [{ spans: [], text: 'Utilize B2B networks', type: 'heading3' }],
  image: {
    alt: 'Placeholder image',
    copyright: null,
    dimensions: { height: 500, width: 900 },
    url: 'https://images.prismic.io/ample-prismic-demo/8d1d6aee-977a-458f-8380-d8bbfa88f00f_wexor-tmg-L-2p8fapOA8-unsplash.jpg?auto=compress,format',
  },
  textAlignment: 'left',
  url: { link_type: 'Web', url: 'http://twitter.com' },
  variation: 'withButton',
}

const buttonLabel = {
  buttonLabel: [
    {
      spans: [],
      text: 'Et excepteur et est et velit est magna id ad id.',
      type: 'paragraph',
    },
  ],
}

// ---------------------------------------------------------

export default {
  _props: {
    ...card,
    ...buttonLabel,
  },
  themeDefault: {
    ...card,
    variation: 'default',
  },
  themeWithButton: {
    ...card,
    ...buttonLabel,
  },
}
