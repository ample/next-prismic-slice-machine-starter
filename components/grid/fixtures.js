// ---------------------------------------------------------

const child = (
  <div style={{ backgroundColor: 'lightgrey', padding: '20px' }}>
    <h2>Lorem ipsum dolor sit amet</h2>
    <p>
      Consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
      dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation.
    </p>
  </div>
);

// ---------------------------------------------------------

export default {
  _props: {
    children: [
      {
        ...child,
      },
      {
        ...child,
      },
      {
        ...child,
      },
    ],
    gap: '20',
    layout: '1/2',
  },
};
