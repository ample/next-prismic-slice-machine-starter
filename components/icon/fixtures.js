export default {
  componentProps: {
    name: 'filters-like-outline',
  },
};
