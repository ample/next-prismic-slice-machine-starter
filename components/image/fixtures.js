// ---------------------------------------------------------

const alt =
  'alt text helps screen-reading tools describe images to visually impaired readers'
const src =
  'https://images.unsplash.com/photo-1522878129833-838a904a0e9e?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2670&q=80'

// ---------------------------------------------------------

const layoutIsNotFill = {
  alt: alt,
  height: 670,
  src: src,
  width: 1004,
}

// ---------------------------------------------------------

export default {
  _props: {
    ...layoutIsNotFill,
  },
  layoutIsFill: {
    alt: alt,
    layout: 'fill',
    objectFit: 'contain',
    src: src,
  },
  layoutIsFixed: {
    layout: 'fixed',
    ...layoutIsNotFill,
  },
  layoutIsIntrinsic: {
    ...layoutIsNotFill,
  },
  layoutIsResponsive: {
    layout: 'responsive',
    ...layoutIsNotFill,
  },
  withLink: {
    ...layoutIsNotFill,
    url: {
      url: 'https://www.helloample.com/',
    },
  },
  withSVG: {
    src: '/storybook/ample-logo.svg',
  },
}
