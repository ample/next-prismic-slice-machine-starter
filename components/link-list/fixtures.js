export default {
  _props: {
    links: [
      { label: 'Vestibulum vitae', url: '#' },
      { label: 'Pellentesque tempor venenatis', url: '#' },
      { label: 'Fusce molestie', url: '#' },
      { label: 'Etiam in lacinia dolor', url: '#' },
    ],
  },
};
