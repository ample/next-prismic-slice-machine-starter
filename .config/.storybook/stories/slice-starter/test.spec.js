// ---------------------------------------------------------
// TODO: REMOVE the "FPO snapshot test" below and add additional tests.
// ----  Snapshots tests are not necessary because theyâ€™re created automatically with Storyshots.
// ---------------------------------------------------------
// ----  If tests are not required, remove this file.
// ---------------------------------------------------------

import renderer from 'react-test-renderer'

// ---------------------------------------------------------

import SliceName from '.'

// ---------------------------------------------------------

import fixtures from './fixtures'

// ---------------------------------------------------------

describe('SliceName', () => {
  it('FPO snapshot', () => {
    const tree = renderer.create(<SliceName {...fixtures._props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
