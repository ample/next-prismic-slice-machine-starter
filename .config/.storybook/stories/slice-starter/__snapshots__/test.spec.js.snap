// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliceName FPO snapshot 1`] = `
<a
  className="slice_name"
  href="url"
>
  <h2>
    name of prop content goes here.
  </h2>
</a>
`;
